### ISSUE TITLE
Some sort of race condition

### ISSUE THREAD

**Issue by balegas** [electric-sql/electric-next#210]
I'm using Linearlite data set with 10K rows and persistent storage.

Start a fresh server:
```bash
rm -fr ./shapes && iex -S mix
```

Make two requests very fast
```bash
http -h GET http://localhost:3000/v1/shape/issue offset==-1 &
http -h GET http://localhost:3000/v1/shape/issue offset==-1 &
```

```

10:01:39.956 [info] GET /v1/shape/issue

10:01:39.956 [info] GET /v1/shape/issue

10:01:39.958 [info] Query String: offset=-1

10:01:39.958 [info] Query String: offset=-1

10:01:39.975 [debug] Table {"public", "issue"} found with 9 columns

10:01:39.982 [debug] Returning shape id 44524560-1722330099976 for shape Shape.new!("public.issue")

10:01:39.993 [error] ** (Plug.Conn.WrapperError) ** (ArgumentError) errors were found at the given arguments:

  * 1st argument: not a list

    :erlang.length({{{"public", "issue"}, :columns}, [%{name: "id", type: "uuid", formatted_type: "uuid", type_mod: -1, pk_position: 0, type_id: {2950, -1}, array_dimensions: 0, array_type: nil}, %{name: "title", type: "text", formatted_type: "text", type_mod: -1, pk_position: nil, type_id: {25, -1}, array_dimensions: 0, array_type: nil}, %{name: "description", type: "text", formatted_type: "text", type_mod: -1, pk_position: nil, type_id: {25, -1}, array_dimensions: 0, array_type: nil}, %{name: "priority", type: "text", formatted_type: "text", type_mod: -1, pk_position: nil, type_id: {25, -1}, array_dimensions: 0, array_type: nil}, %{name: "status", type: "text", formatted_type: "text", type_mod: -1, pk_position: nil, type_id: {25, -1}, array_dimensions: 0, array_type: nil}, %{name: "modified", type: "timestamptz", formatted_type: "timestamp with time zone", type_mod: -1, pk_position: nil, type_id: {1184, -1}, array_dimensions: 0, array_type: nil}, %{name: "created", type: "timestamptz", formatted_type: "timestamp with time zone", type_mod: -1, pk_position: nil, type_id: {1184, -1}, array_dimensions: 0, array_type: nil}, %{name: "kanbanorder", type: "text", formatted_type: "text", type_mod: -1, pk_position: nil, type_id: {25, -1}, array_dimensions: 0, array_type: nil}, %{name: "username", type: "text", formatted_type: "text", type_mod: -1, pk_position: nil, type_id: {25, -1}, array_dimensions: 0, array_type: nil}]})
    (electric 0.2.1) lib/electric/shapes/shape.ex:74: Electric.Shapes.Shape.load_column_info/2
    (electric 0.2.1) lib/electric/shapes/shape.ex:50: Electric.Shapes.Shape.new/2
    (electric 0.2.1) lib/electric/plug/serve_shape_plug.ex:97: Electric.Plug.ServeShapePlug.Params.cast_root_table/2
    (electric 0.2.1) lib/electric/plug/serve_shape_plug.ex:39: Electric.Plug.ServeShapePlug.Params.validate/2
    (electric 0.2.1) lib/electric/plug/serve_shape_plug.ex:134: Electric.Plug.ServeShapePlug.validate_query_params/2
    (electric 0.2.1) lib/electric/plug/serve_shape_plug.ex:1: Electric.Plug.ServeShapePlug.plug_builder_call/2
    (electric 0.2.1) deps/plug/lib/plug/router.ex:246: anonymous fn/4 in Electric.Plug.Router.dispatch/2
    (telemetry 1.2.1) /Users/vbalegas/workspace/electric-benchmarking/electric-next/packages/sync-service/deps/telemetry/src/telemetry.erl:321: :telemetry.span/3
    (electric 0.2.1) deps/plug/lib/plug/router.ex:242: Electric.Plug.Router.dispatch/2
    (electric 0.2.1) lib/electric/plug/router.ex:1: Electric.Plug.Router.plug_builder_call/2
    (bandit 1.5.5) lib/bandit/pipeline.ex:124: Bandit.Pipeline.call_plug!/2
    (bandit 1.5.5) lib/bandit/pipeline.ex:36: Bandit.Pipeline.run/4
    (bandit 1.5.5) lib/bandit/http1/handler.ex:12: Bandit.HTTP1.Handler.handle_data/3
    (bandit 1.5.5) lib/bandit/delegating_handler.ex:18: Bandit.DelegatingHandler.handle_data/3
    (bandit 1.5.5) /Users/vbalegas/workspace/electric-benchmarking/electric-next/packages/sync-service/deps/thousand_island/lib/thousand_island/handler.ex:411: Bandit.DelegatingHandler.handle_continue/2
    (stdlib 6.0) gen_server.erl:2163: :gen_server.try_handle_continue/3
    (stdlib 6.0) gen_server.erl:2072: :gen_server.loop/7
    (stdlib 6.0) proc_lib.erl:329: :proc_lib.init_p_do_apply/3

```

**Comment by kevin-dp** [comment]
@robacourt This seems very similar to the server-side error of the 500 problem i linked to you this morning. What do you think?

**Comment by kevin-dp** [comment]
@balegas I can reproduce this issue

**Comment by robacourt** [comment]
> @robacourt This seems very similar to the server-side error of the 500 problem i linked to you this morning. What do you think?

Yeah, this is the same issue. @balegas the fix is here: #214

**Comment by robacourt** [comment]
Fixed with #214
